heareval/predictions/task_predictions.py

```diff
@@ -253,6 +253,38 @@ def validation_step(self, batch, batch_idx):
     def test_step(self, batch, batch_idx):
         return self._step(batch, batch_idx)
 
+    def log_scores(self, name: str, score_args):
+        """Logs the metric score value for each score defined for the model"""
+        assert hasattr(self, "scores"), "Scores for the model should be defined"
+        end_scores = {}
+        for score in self.scores:
+            score_ret = score(*score_args)
+            validate_score_return_type(score_ret)
+            # If the returned score is a tuple, store each subscore as separate entry
+            if isinstance(score_ret, tuple):
+                # The first score in the returned tuple will be used for
+                # any optimisation criterion, for instance if the metric is
+                # primary metric, the score corresponding to the first tuple
+                # will be used for early stopping
```

Comment:
This still isn't right. The first score in the returned tuple for the first self.scores is the optimization criterion.

The comment suggests that the first score in the returned tuple for EVERY self.scores is the optimization criterion